addons/thtracker/thtracker.lua

```diff
@@ -24,162 +24,83 @@
-mob = nil
-count = 0
-
-local settingtab = nil
-local settings_file = 'data\\settings.xml'
-local settingtab = config.load(settings_file)
-if settingtab == nil then
-	write('No settings file found. Ensure you have a file at data\\settings.xml')
-end
-
-if settingtab['posx'] ~= nil then
-	posx = settingtab['posx']
-	posy = settingtab['posy']
-end	
-	
-function initialize()
-	--[[windower.text.create(box_name)
-	windower.text.set_bg_color(box_name,200,30,30,30)
-	windower.text.set_color(box_name,255,200,200,200)
-	windower.text.set_location(box_name,posx,posy)
-	windower.text.set_visibility(box_name, see)
-	windower.text.set_bg_visibility(box_name,1)
-	windower.text.set_font(box_name,'Arial',12)
-	windower.text.set_text(box_name, thoutput)]]
-	--log('Initializing TH Tracker')
-	t = texts.new('test text box')
-	thbox = texts.new('${boxtext}', {
-		pos = {
-			x = posx,
-			y = posy
-		},
-		bg = {
-			visible = true,
-			alpha = 255,
-			red = 30,
-			green = 30,
-			blue = 30
-		},
-		text = {
-			size = 10,
-			alpha = 200,
-			red = 200,
-			green = 200,
-			blue = 200
-		}
-	})
-	thbox.boxtext = thoutput
-	thbox:show()
-end
-
-function dispose()
-    windower.text.delete(box_name)
-end
-
-function thunload()
-	dispose()
-	windower.send_command('unalias th')
-end 
-
-windower.register_event('load', function(...)
-	windower.send_command('alias th lua c thtracker')
-	
-	initialize()
+defaults = {}
+defaults.pos = {}
+defaults.pos.x = 1000
+defaults.pos.y = 200
+defaults.color = {}
+defaults.color.alpha = 200
+defaults.color.red = 200
+defaults.color.green = 200
+defaults.color.blue = 200
+defaults.bg = {}
+defaults.bg.alpha = 200
+defaults.bg.red = 30
+defaults.bg.green = 30
+defaults.bg.blue = 30
+
+settings = config.load(defaults)
+
+th = texts.new(settings)
+
+windower.register_event('addon command', function(command, ...)
+    command = command and command:lower()
+    local args = {...}
+
+    if command == 'pos' then
+        local posx, posy = tonumber(params[2]), tonumber(params[3])
+        if posx and posy then
+            th:pos(posx, posy)
+        end
+    elseif command == "hide" then
+        th:hide()
+    elseif command == 'show' then
+        th:show()
+    else
+        print('th help : Shows help message')
+        print('th pos <x> <y> : Positions the list')
+        print('th hide : Hides the box')
+        print('th show : Shows the box')
+    end
 end)
 
-windower.register_event('login', initialize)
-windower.register_event('logout', dispose)
-windower.register_event('unload', thunload)
-
-
-windower.register_event('addon command', function(...)
-	local params = {...};
-	if #params < 1 then
-		return
+windower.register_event('incoming text', function(original, new, color)
+	original = original:strip_format()
+	if string.find(original, "Treasure Hunter") and not string.find(original, "TH Tracker:") then
+		name, count = original:match('Additional effect: Treasure Hunter effectiveness against the (.*) increases to (%d+)')
```

Comment:
I'm actually in the middle of correcting the issue with the mob being killed at least having the same name. I want to eventually do better matching to support multiple targets, but that is beyond me right now. I should have the scopes corrected shortly.